### PR TITLE
acid code cleanup

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Tested, but TM first in case I overlooked something.

Refactors a bit of acid code to be less horrible.

Its still kinda horrible, but less snowflake and funny atom var.

Mainly things are now done by atom procs as appropriate, and with signals for some behaviors instead of funny checks during process.
## Why It's Good For The Game
Better code.
## Changelog
:cl:
code: Cleaned up some acid code
/:cl:

### CODE REVIEW

File: code/game/objects/effects/aliens.dm [hunk 1]
```diff
@@ -137,70 +137,44 @@
...
 
-/obj/effect/xenomorph/acid/strong
-	name = "strong acid"
-	acid_strength = STRONG_ACID_STRENGTH
-	acid_damage = 175
-	icon_state = "acid_strong"
+	var/obj/effect/xenomorph/acid/current_acid = target.get_self_acid()
+	if(current_acid)
+		ticks += current_acid.ticks
+		qdel(current_acid)
```
Comment (TiviPlus) [inline code/game/objects/effects/aliens.dm:148]: Return initialize qdel instead?

File: code/game/objects/effects/aliens.dm [hunk 2]
```diff
@@ -211,6 +185,60 @@
...
+	SIGNAL_HANDLER
+	if(!ishuman(user))
+		qdel(src)
+		return
+	INVOKE_ASYNC(src, PROC_REF(on_pickup), source, user)
+
+///Sig handler to show this acid is attached to something
+/obj/effect/xenomorph/acid/proc/on_pickup(obj/item/item, mob/living/carbon/human/human_user)
+	human_user.visible_message(span_danger("Corrosive substances seethe all over [human_user] as it retrieves the acid-soaked [item]!"),
+	span_danger("Corrosive substances burn and seethe all over you upon retrieving the acid-soaked [item]!"))
```
Comment (TiviPlus) [inline code/game/objects/effects/aliens.dm:214]: This is super long and should use the pronoun helpers

File: code/game/objects/effects/aliens.dm [hunk 3]
```diff
@@ -211,6 +185,60 @@
...
+	playsound(human_user, SFX_ACID_HIT, 25)
+	human_user.emote("pain")
+	var/list/affected_limbs = list("l_hand", "r_hand", "l_arm", "r_arm")
+	var/limb_count = null
+	for(var/datum/limb/limb in human_user.limbs)
```
Comment (TiviPlus) [inline code/game/objects/effects/aliens.dm:219]: As?

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Some tochat could be replaced with ballons but looks good enough

**Comment by Lumipharon** [comment]
updated

**Comment by Lumipharon** [comment]
@TiviPlus should be good to merge now

**Comment by MiniMeatwad** [comment]
@Lumipharon Acid does not disappear if you pick up an item.

**Pull request merged**